### DISCUSSION TITLE
React native support for react-markdown?

### DISCUSSION THREAD

**Discussion by loredanacirstea** [remarkjs/remark#579] [Q&A] [answered]
I am trying to use `react-markdown` in React Native with custom renderers, but I get a blank page (I substituted all default renderers with RN components). Has anyone else tried it? - should it work?

```
import {Text} from 'react-native';
import ReactMarkdown from 'react-markdown';
import gfm from 'remark-gfm';

const renderers = {
    text: ({value}) => {
        return <Text>value</Text>;
    },
    // link, etc.
}

const markdown = `Just a link: https://reactjs.com.`

class Custom extends Component{
    render() {
        return (
            <ReactMarkdown plugins={[gfm]} children={markdown} renderers={renderers} />
        );
    }
}
```

**Comment by ChristianMurphy** [comment, marked as answer]
It is theoretically possible (see https://github.com/remarkjs/react-markdown/issues/141#issuecomment-364823198).
No one has document/shared an example of using it with RN yet, a PR adding docs would be welcome!

**Comment by imwithye** [comment]
I tried various Markdown rendering solutions in React Native but couldn’t find one that looked elegant or felt flexible enough. So I built react-native-remark.

https://github.com/imwithye/react-native-remark

react-native-remark:

• Built on top of the unified and remark ecosystems
• Supports GitHub Flavored Markdown (GFM), including tables, code blocks, images, and lists
• Includes syntax highlighting for code blocks with dark mode support
• Provides horizontal scroll for tables to improve readability on mobile
• Enables custom renderers and styles for full UI customization
• Exposes callbacks for handling link presses and code copy actions
• Designed for elegant, native-feeling Markdown rendering in React Native apps

**Reply by fazi-gondal** [reply]
@imwithye  Nice work sir please add more features like Latex and mermaid support, GitHub callout, and html tags support use rehype raw plugin

**Reply by Nedi11** [reply]
Latex support would be nice
